arclight-common/src/main/java/io/izzel/arclight/common/mixin/core/network/handshake/client/CHandshakePacketMixin.java

```diff
@@ -37,15 +41,21 @@ public class CHandshakePacketMixin {
         if (SpigotConfig.bungee && !Objects.equals(this.fmlVersion, FMLNetworkConstants.NETVERSION)) {
             String[] split = this.ip.split("\0");
             if (split.length == 4) {
-                Property[] properties = GSON.fromJson(split[3], Property[].class);
-                for (Property property : properties) {
-                    if (Objects.equals(property.getName(), EXTRA_DATA)) {
-                        String extraData = property.getValue().replace("\1", "\0");
+                PropertyMap properties = GSON.fromJson(split[3], PropertyMap.class);
+                for (Map.Entry<String, Property> property : properties.entries()) {
+                    if (property.getKey().equalsIgnoreCase("forgeClient")) {
+                        if (property.getValue().getValue().equalsIgnoreCase(FMLNetworkConstants.NETVERSION)) {
+                            this.fmlVersion = FMLNetworkConstants.NETVERSION;
+                        }
+                    }
+
+                    if (Objects.equals(property.getValue().getName(), EXTRA_DATA)) {
+                        String extraData = property.getValue().getValue().replace("\1", "\0");
```

Comment:
Why there're irrelevant codes in this pr?

Reply:
Oh apologies I forgot to remove this. This is the logic for the client reset packet handling (and Velocity support). Will quickly revert this part now